src/main/java/com/github/legoatoom/connectiblechains/ConnectibleChains.java

```diff
@@ -61,6 +71,30 @@
      */
     public static ModConfig runtimeConfig;
 
+    /**
+     * Here is where the fun begins.
+     */
+    @Override
+    public void onInitialize() {
+        ModEntityTypes.init();
+        ChainTypesRegistry.init();
+        BuiltinCompat.init();
+
+        AutoConfig.register(ModConfig.class, Toml4jConfigSerializer::new);
+        ConfigHolder<ModConfig> configHolder = AutoConfig.getConfigHolder(ModConfig.class);
+        fileConfig = configHolder.getConfig();
+        runtimeConfig = new ModConfig().copyFrom(fileConfig);
+
+        UseBlockCallback.EVENT.register(ConnectibleChains::chainUseEvent);
+
+        ServerPlayConnectionEvents.INIT.register((handler, server) -> fileConfig.syncToClient(handler.getPlayer()));
+
+        if (FabricLoader.getInstance().getEnvironmentType() == EnvType.SERVER) {
+            // ClientInitializer uses CLIENT_STARTED
+            ServerLifecycleEvents.SERVER_STARTED.register((server) -> ChainTypesRegistry.lock());
+        }
```

Comment:
You sure that this doesn't cause client / server issues?

Reply:
I didn't know of DedicatedServerInitializer when I wrote this code, moved it.